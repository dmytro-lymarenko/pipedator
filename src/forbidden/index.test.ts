import { forbidden } from './index';

describe('forbidden()', () => {
	it('should validate undefined', () => {
		expect(forbidden().validate(undefined)).toBeNull();
	});

	it('should fail', () => {
		expect(forbidden().validate(1)).toMatchSnapshot();
		expect(forbidden().validate(null)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(forbidden('Custom message').validate(null)).toMatchSnapshot();
	});
});
