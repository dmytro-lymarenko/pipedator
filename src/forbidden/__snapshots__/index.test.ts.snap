// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forbidden() should fail 1`] = `
Object {
  "children": Array [],
  "message": "Value should equal to undefined",
  "path": "",
}
`;

exports[`forbidden() should fail 2`] = `
Object {
  "children": Array [],
  "message": "Value should equal to undefined",
  "path": "",
}
`;

exports[`forbidden() should support custom message 1`] = `
Object {
  "children": Array [],
  "message": "Custom message",
  "path": "",
}
`;
